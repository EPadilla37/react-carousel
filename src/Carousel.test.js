import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Carousel from "./Carousel";
import Card from "./Card";

it("works when you click on the right arrow", function() {
  const { queryByTestId, queryByAltText } = render(<Carousel />);

  // expect the first image to show, but not the second
  expect(queryByAltText("Photo by Richard Pasquarella on Unsplash")).toBeInTheDocument();
  expect(queryByAltText("Photo by Pratik Patel on Unsplash")).not.toBeInTheDocument();

  // move forward in the carousel
  const rightArrow = queryByTestId("right-arrow");
  fireEvent.click(rightArrow);

  // expect the second image to show, but not the first
  expect(queryByAltText("Photo by Richard Pasquarella on Unsplash")).not.toBeInTheDocument();
  expect(queryByAltText("Photo by Pratik Patel on Unsplash")).toBeInTheDocument();
});

// Smoke test for Card component
test("Card component renders without crashing", () => {
  render(<Card caption="Test Caption" src="test.jpg" currNum={1} totalNum={3} />);
});

// Snapshot test for Card component
test("Card component matches snapshot", () => {
  const { container } = render(<Card caption="Test Caption" src="test.jpg" currNum={1} totalNum={3} />);
  expect(container).toMatchSnapshot();
});

// Smoke test for Carousel component
test("Carousel component renders without crashing", () => {
  render(<Carousel />);
});

// Snapshot test for Carousel component
test("Carousel component matches snapshot", () => {
  const { container } = render(<Carousel />);
  expect(container).toMatchSnapshot();
});

// Test for left arrow behavior
it("left arrow moves to previous image", function() {
  const { queryByTestId, queryByAltText } = render(<Carousel />);
  const rightArrow = queryByTestId("right-arrow");
  fireEvent.click(rightArrow);
  const leftArrow = queryByTestId("left-arrow");
  fireEvent.click(leftArrow);
  expect(queryByAltText("Photo by Richard Pasquarella on Unsplash")).toBeInTheDocument();
});

// Test for arrow visibility
it("left arrow is missing on first image, and right arrow is missing on last image", function() {
  const { queryByTestId } = render(<Carousel />);
  const leftArrow = queryByTestId("left-arrow");
  expect(leftArrow).toHaveClass("hidden");

  const rightArrow = queryByTestId("right-arrow");
  fireEvent.click(rightArrow);
  expect(leftArrow).not.toHaveClass("hidden");
});
