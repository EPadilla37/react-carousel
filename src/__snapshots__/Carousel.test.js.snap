// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component matches snapshot 1`] = `
<div>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      Test Caption
    </h4>
    <img
      alt="Test Caption"
      class="Card-image"
      src="test.jpg"
    />
    <small
      class="Card-small"
    >
      Image 
      1
       of 
      3
      .
    </small>
  </div>
</div>
`;

exports[`Carousel component matches snapshot 1`] = `
<div>
  <div
    class="Carousel"
  >
    <h1>
      Shells from far away beaches.
    </h1>
    <div
      class="Carousel-main"
    >
      <i
        class="fas fa-chevron-circle-left fa-2x hidden"
        data-testid="left-arrow"
      />
      <div
        class="Card"
      >
        <h4
          class="Card-title"
        >
          Photo by Richard Pasquarella on Unsplash
        </h4>
        <img
          alt="Photo by Richard Pasquarella on Unsplash"
          class="Card-image"
          src="image1.jpg"
        />
        <small
          class="Card-small"
        >
          Image 
          1
           of 
          3
          .
        </small>
      </div>
      <i
        class="fas fa-chevron-circle-right fa-2x "
        data-testid="right-arrow"
      />
    </div>
  </div>
</div>
`;
